### PR TITLE
[ROU-3970]: TimePicker/MonthPicker - Accessibility and other improvements.

### PR DESCRIPTION
**Pull request by joselrio**
This PR is for implement several improvements to:
- Date, Time and Month pickers in terms of accessibility, setting the initial value to the "hidden" input accordingly and updating those values at the OnSelected date/time/moth as well;
- Improved accessibility highlighted state to the DropdownServerSide items;
- Redefined the way onOpen and onClose configs was being set to the Time and Month picker;
- Implemented a validation to ensure OnInitialized callback event is triggered only once at the pattern initialisation;

### CODE REVIEW

**Review by JoaoFerreira-FrontEnd — DISMISSED** [review]

File: src/scripts/Providers/OSUI/Timepicker/Flatpickr/FlatpickrTime.ts [hunk 1]
```diff
@@ -158,36 +160,39 @@ namespace Providers.OSUI.TimePicker.Flatpickr {
 			this.triggerPlatformInitializedEventCallback();
 		}
 
+		/**
+		 * Method that will be triggerd each time TimePicker will be closed
```
Comment (bmarcelino-fe) [inline src/scripts/Providers/OSUI/Timepicker/Flatpickr/FlatpickrTime.ts:164]: Typo on Triggered

File: src/scripts/Providers/OSUI/Timepicker/Flatpickr/FlatpickrTime.ts [hunk 2]
```diff
@@ -158,36 +160,39 @@ namespace Providers.OSUI.TimePicker.Flatpickr {
...
+		protected onClose(): void {
+			// Check if bodyOnClickEvent exist
+			if (this._bodyOnClickGlobalEvent !== undefined) {
+				// Enabled it since TimePicker has been closed
+				this._bodyOnClickGlobalEvent.enableBodyClickEvent();
+			}
+		}
+
+		/**
+		 * Method that will be triggerd each time TimePicker will open
```
Comment (bmarcelino-fe) [inline src/scripts/Providers/OSUI/Timepicker/Flatpickr/FlatpickrTime.ts:178]: Same typo here

File: src/scripts/OSFramework/OSUI/Pattern/DropdownServerSideItem/scss/_dropdownserversideitem.scss [hunk 3]
```diff
@@ -17,10 +17,12 @@
 	position: relative;
 	transition: background 250ms ease;
 	width: 100%;
+	z-index: 0;
```
Comment (bmarcelino-fe) [inline src/scripts/OSFramework/OSUI/Pattern/DropdownServerSideItem/scss/_dropdownserversideitem.scss:20]: One question, we don't need to use the layer systems here? Why we use the zindex 0 and 1?

**Review by bmarcelino-fe — CHANGES_REQUESTED** [review]

File: src/scripts/OSFramework/OSUI/Pattern/DropdownServerSideItem/scss/_dropdownserversideitem.scss (hunk 3 above)
Comment (joselrio) [reply src/scripts/OSFramework/OSUI/Pattern/DropdownServerSideItem/scss/_dropdownserversideitem.scss:20]: This is the same element...

File: src/scripts/Providers/OSUI/Timepicker/Flatpickr/FlatpickrTime.ts (hunk 1 above)
Comment (joselrio) [reply src/scripts/Providers/OSUI/Timepicker/Flatpickr/FlatpickrTime.ts]: Done!

File: src/scripts/Providers/OSUI/Timepicker/Flatpickr/FlatpickrTime.ts (hunk 2 above)
Comment (joselrio) [reply src/scripts/Providers/OSUI/Timepicker/Flatpickr/FlatpickrTime.ts]: Done!

**Review by JoaoFerreira-FrontEnd — APPROVED** [review]

**Review by bmarcelino-fe — APPROVED** [review]

**Pull request merged**